### ISSUE TITLE
[New resources]: AWS Control Tower introduces APIs to register Organizational Units

### ISSUE THREAD

**Issue by EreminAnton** [hashicorp/terraform-provider-aws#35849]
### Description

If i understand correctly, this functionality exists in the Go SDK v2.
New resources for configuring control tower via terraform. 

### Requested Resource(s) and/or Data Source(s)

EnableBaseline/UpdateEnabledBaseline/DisableBaseline: Take action on a baseline for an OU
ResetEnabledBaseline: Remediate resource drift on an OU with an enabled baseline (including nested/mandatory controls drift)
aws_control_tower_baseline
GetEnabledBaseline/ListEnabledBaselines: Discover configurations for your enabled baselines
GetBaselineOperation: View the status of a particular baseline operation
data.aws_control_tower_baseline.this

GetBaseline/ListBaselines: Discover content of AWS Control Tower managed baselines
data.aws_control_tower_list_baselines.this

### Potential Terraform Configuration

```terraform
resource "aws_control_tower_baseline" {
  target_ou_id = ""
  baselineIdentifier = ""
  baselineVersion = ""
  tags = {}
}
```

### References

https://aws.amazon.com/about-aws/whats-new/2024/02/aws-control-tower-apis-register-organizational-units/

### Would you like to implement a fix?

No

**Comment by albsilv-aws** [comment]
I'll work on this one!

**Comment by sasidhar-aws** [comment]
I am working on this issue.

**Comment by edgy-noodle** [comment]
Hey @sasidhar-aws,
Is this actively being worked on? Are there any progress updates/ETAs?

I will need this functionality for a personal project and so will very likely end up forking the repo and implementing it myself anyway - might as well contribute if this has not been touched yet or is at a pause.

**Comment by sasidhar-aws** [comment]
yes, I have been held with other works. I will close this in next week. thanks.

**Comment by sergiopicca** [comment]
Any update regarding this issue? The pull request is still in draft status. Thanks in advance!

**Comment by sasidhar-aws** [comment]
I am stuck at converting terraform structure to aws object. Autoflex is failing to  convert parameters as Value attribute is defined as document.Interface. I am trying multiple approaches.

```
[ERROR] aws.autoflex: Source does not implement attr.Value: autoflex.target.path=Parameters tf_req_id=a16e6c8c-ed40-7ca9-4613-9fdd3551ded0 tf_resource_type=aws_controltower_baseline tf_provider_addr=registry.terraform.io/hashicorp/aws tf_rpc=ApplyResourceChange autoflex.source.path=Parameters autoflex.target.type=[]github.com/aws/aws-sdk-go-v2/service/controltower/types.EnabledBaselineParameter tf_mux_provider="*proto5server.Server" autoflex.source.type=[]github.com/hashicorp/terraform-provider-aws/internal/service/controltower.parameters
    baseline_test.go:124: Step 1/2 error: Error running apply: exit status 1
        
        Error: Incompatible Types
```
